src/ApiService/ApiService/onefuzzlib/Secrets.cs

```diff
@@ -28,41 +30,51 @@ public class SecretsOperations : ISecretsOperations {
         _config = config;
     }
 
-    public (Uri, string) ParseSecretUrl(Uri secretsUrl) {
+    public static (Uri, string) ParseSecretUrl(Uri secretsUrl) {
         // format: https://{vault-name}.vault.azure.net/secrets/{secret-name}/{version}
         var vaultUrl = $"{secretsUrl.Scheme}://{secretsUrl.Host}";
         var secretName = secretsUrl.Segments[^2].Trim('/');
         return (new Uri(vaultUrl), secretName);
     }
 
-    public virtual async Task<SecretData<T>> SaveToKeyvault<T>(SecretData<T> secretData) {
-
-        if (secretData.Secret is SecretAddress<T> secretAddress) {
+    public virtual async Task<SecretData<T>> StoreSecretData<T>(SecretData<T> secretData) {
+        if (secretData.Secret.IsHIddden) {
             return secretData;
-        } else if (secretData.Secret is SecretValue<T> sValue) {
-            var secretName = Guid.NewGuid();
-            string secretValue;
-            if (sValue.Value is string secretString) {
-                secretValue = secretString.Trim();
-            } else {
-                secretValue = JsonSerializer.Serialize(sValue.Value, EntityConverter.GetJsonSerializerOptions());
-            }
-
-            var kv = await StoreInKeyvault(GetKeyvaultAddress(), secretName.ToString(), secretValue);
-            return new SecretData<T>(new SecretAddress<T>(kv.Id));
         }
+        var kv = await StoreSecret(secretData.Secret);
+        return new SecretData<T>(new SecretAddress<T>(kv));
+    }
```

Comment:
We don't need this because of the default implementation in `ISecretsOperations` right?